### PR TITLE
app: [iOS] add ViewEvent

### PR DESCRIPTION
**Pull request by Inkeliz**
Fix #305

Signed-off-by: Inkeliz <inkeliz@inkeliz.com>

### CODE REVIEW

File: app/os_ios.go [hunk 1]
```diff
@@ -86,10 +86,17 @@ import (
 	"gioui.org/unit"
 )
 
-type ViewEvent struct{}
+type ViewEvent struct {
+	// View is a CFTypeRef for the NSView for the window.
+	View uintptr
+
+	// Window is a CFTypeRef for the NSView for the window.
+	Window uintptr
```
Comment (eliasnaur) [inline app/os_ios.go:94]: Window is a confusing name for a NSView. Can't client code fetch the parent NSView (or NSWindow) through ViewEvent.View?

Comment (Inkeliz) [reply app/os_ios.go]: I think it's possible. At least, using `[view superview]` seems to work. _Considering that `view` is the C.CFTypeRef  of `ViewEvent.View`._

**Comment by eliasnaur** [comment]
> I my case I use:
> [[view superview] addSubview:myCustomView];

This is too brittle; if we ever drop the parent view, your code will silently break.
Can you confirm something like https://developer.apple.com/documentation/appkit/nswindow/1419160-contentview works too?

**Comment by Inkeliz** [comment]
There's no `contentView` on UIView (https://developer.apple.com/documentation/uikit/uiview?language=objc). It gives a error, "no visible @interface for UIView".

----

In the previous PR (with View and "Window"), both are UIView, however one of them is "GioView" and another one is the "RootView" (I don't know the name of that, but is here GioView is added to):

https://github.com/gioui/gio/blob/d2dc0d52495efc6ddb476745638fa162205bba04/app/os_ios.m#L24-L33

**Comment by eliasnaur** [comment]
I confused UIView and NSView, sorry.

I'm still worried we can't ever remove the parent view. Why doesn't `[view addSubView:...]` directly on the ViewEvent.View work?

**Comment by Inkeliz** [comment]
Using `[view addSubView:...]` works too. 🎉

**Comment by eliasnaur** [comment]
Thanks for confirming. Can you add the missing empty ViewEvent in onDestroy (see os_android.go for an example)? Otherwise, I think the change is good to go.

**Comment by Inkeliz** [comment]
I add the `w.w.Event(ViewEvent{})` on `onDestroy`, but I don't think it makes sense. On iOS the `onDestroy` sends `system.DestroyEvent`, where Android doesn't. 🤔

**Comment by eliasnaur** [comment]
> I add the `w.w.Event(ViewEvent{})` on `onDestroy`, but I don't think it makes sense. On iOS the `onDestroy` sends `system.DestroyEvent`, where Android doesn't. thinking

Android is special in that the GioView may be destroyed and recreated, for example during rotations. And yes, the Android backend should probably send a DestroyEvent when the GioView is gone for good.

However, the empty ViewEvent merely notifies the Gio program that the view handle is no longer valid, which is not the same as the app.Window being destroyed.

I've merged your patch with a short description in the commit message; Chris and I want to improve commit messages so they can be understood by casual Gio users, and to automate the newsletter change logs.

**Pull request closed without merging**